src/server.cpp

```diff
@@ -798,10 +800,16 @@ static CTimingMeas JitterMeas ( 1000, "test2.dat" ); JitterMeas.Measure(); // TE
     bool bUpdateChannelLevels      = false;
     bool bSendChannelLevels        = false;
 
+    // flip to the idle page for the next call, in case it overlaps with the previous
+    iChannelDataPage = 1-iChannelDataPage;
```

Comment:
I was also thinking of using a similar implementation as your "page". But this only makes sense if the decoding is done in one thread. The idea was that if the single decoding thread requires the complete time, the mixing/encoding could be done in parallel when the next frame is decoded. But as far as I understood, you now also spilt the decoding in multiple threads. Then it does not make any more sense to use the pages. You then use 100 % of the CPU for decoding and after that 100 % of the CPU for mixing and encoding. So you would not gain anything from doing decoding and mixing/encoding in parallel. Do you agree?

Reply:
The FutureSync no longer occurs after the max threads are started, so they can continue to run after the OnTimer function has returned. That's when you get concurrency between decode and mix.

Instead, the FutureSync is called just _before_ launching new mix threads. So at that point it will block waiting for the last timer to completely finish before starting new mixes.